output/file.go

```diff
@@ -0,0 +1,103 @@
+package output
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/vbauerster/mpb/v5"
+	"github.com/vbauerster/mpb/v5/decor"
+)
+
+type FileWriter struct {
+	FullPath string
+	Filename string
+}
+
+func NewFileWriter(url *url.URL, options *Options) *FileWriter {
+	var tempPath string
+
+	if options.OutputFile == "" {
+		tempFile := filepath.Base(url.Path)
+		tempPath = fmt.Sprintf("./%s", tempFile)
+	} else {
+		tempPath = options.OutputFile
+	}
+
+	fullPath := tempPath
+
+	if !options.Overwrite {
+		fullPath = checkFile(tempPath)
+	}
+
+	outFile := filepath.Base(fullPath)
+
+	return &FileWriter{
+		FullPath: fullPath,
+		Filename: outFile,
+	}
+}
+
+func checkFile(path string) string {
```

Comment:
The name `checkFile` did not give me enough information about this function actually doing.
This function's responsibility is giving a new name that does not overlap with existing files.
I think `makeNonOverlappingFilename` might be more informative.

Reply:
Done.